File: 392.php
<?php

class Solution {

    /**
     * @param String $s
     * @param String $t
     * @return Boolean
     */
    function isSubsequence($s, $t) {
        $l1 = strlen($s);
        $l2 = strlen($t);

        if ($l1 == 0) return true;

        $is = 0;
        $it = 0;

        while ($it < $l2) {
            if ($s[$is] == $t[$it]) {
                $it++;
                if (++$is == $l1) {
                    return true;
                }
            } else {
                $it++;
            }
        }
        return false;
    }
}

$solution = new Solution();
var_dump($solution->isSubsequence('abc', 'ahbgdc'));

File: go/greedy/392.go
package main

func isSubsequence(s string, t string) bool {
	sI, tI, sL, tL := 0, 0, len(s), len(t)
	if sL == 0 {
		return true
	}

	for sI < sL {
		if tI == tL {
			return false
		}
		if s[sI] == t[tI] {
			sI++
		}
		if sI == sL {
			return true
		}
		tI++
	}

	return false
}
